src/utils/hooks/useRestartReview.ts

```diff
@@ -25,23 +31,36 @@ const useRestartReview: UseRestartReview = ({ reviewId, structure, assignment })
 
   const constructReviewPatch: ConstructReviewPatch = (structure) => {
     const elements = Object.values(structure?.elementsById || {})
-
+    // Check for draft review response in case consolidation is ongoing and new review response is added
+    const isDraftReviewResponse = (element: PageElement) =>
+      element.thisReviewLatestResponse &&
+      element.thisReviewLatestResponse.status === ReviewResponseStatus.Draft
     // Exclude not assigned, not visible and missing responses
     const reviewableElements = elements.filter(
-      (element) => element?.isAssigned && element?.element.isVisible && element.response?.id
+      (element) => element.isPendingReview && !isDraftReviewResponse(element)
     )
 
     // For re-assignment this would be slightly different, we need to consider latest review response of this level
     // not necessarily this thisReviewLatestResponse (would be just latestReviewResponse, from all reviews at this level)
     const reviewResponseCreate = reviewableElements.map(
-      ({ thisReviewLatestResponse, response, assignmentId }) => {
-        // create new if application response is not linked to latest review response
-        const shouldCreateNew = thisReviewLatestResponse?.applicationResponse?.id !== response?.id
+      ({
+        isPendingReview,
+        thisReviewLatestResponse,
+        response,
+        reviewQuestionAssignmentId,
+        latestPreviousLevelReviewResponse,
+      }) => {
+        const applicationResponseId = assignment.level > 1 ? undefined : response?.id
+        const reviewResponseLinkId =
+          assignment.level === 1 ? undefined : latestPreviousLevelReviewResponse?.id
+        // create new if element is awaiting review
+        const shouldCreateNew = isPendingReview
```

Comment:
Hmm I'm not sure if I got this straight, isn't all elements in `reviewableElements` already have `isPendingReview` = true. So why do this check `shouldCreateNew ? null : thisReviewLatestResponse?.decision` ?

Reply:
I think this is an issue, thanks for pointing it out

Reply:
I had to create some diagrams to make sense of it all again:

it's this one that address this comment: https://app.diagrams.net/#G1ttYeAHmdxWCnxyTY5UX1VSRDFPe-5Nvd

And here are the changes:

f72b56f838d337fd9e714cf67cbed499edf22eb0

More diagrams:

https://app.diagrams.net/#G1345qSOpVZ3PIw-_YXsZg_qgvm8CmIMDy
https://app.diagrams.net/#G1XyiT_SSpjy521KJA-hJ8WjGgdqj8m5Ru
https://app.diagrams.net/#G1JmZc2daHZswUyjOFWZicJy8sttpS_-eZ

Reply:
This is great. I'm trying to read those, ideally next step would be to have it updated (on top of existing PRs) to have it on our wiki for future reference. Thanks for doing this, it will probably take me half-hour to read this and make an exact idea of how this is working, but that's for a good sake of understanding and making sure there isn't any gaps to this logic. I'll make any comments and questions direct to you in Telegram so this can be more straight-forward.